crates/extricrate/src/lib.rs

```diff
@@ -44,13 +61,86 @@ pub mod dependencies {
 
     pub type UseStatements = Vec<UseStatement>;
 
+    #[derive(Debug, Hash, PartialEq, Eq)]
     pub struct File(String);
 
     pub type UseStatementMap = HashMap<File, UseStatements>;
 
+    struct UseVisitor;
+
+    impl<'ast> Visit<'ast> for UseVisitor {
+        fn visit_item_use(&mut self, node: &'ast ItemUse) {
+            let tokens = node.to_token_stream();
+            dbg!(flatten_use_tree("", &node.tree));
+            visit::visit_item_use(self, node);
+        }
+    }
+
+    fn flatten_use_tree(prefix: &str, tree: &UseTree) -> Vec<UseStatementType> {
+        match tree {
+            UseTree::Path(UsePath { ident, tree, .. }) => {
+                let new_prefix = if prefix.is_empty() {
```

Comment:
Can you extract a lambda here? Something like "prefixed" that applies the prefix to the given str